tests/test_spec.py

```diff
@@ -178,6 +178,29 @@ def test_get(val):
             "schema": {"$ref": "#/components/schemas/PaginationMetadata"},
         }
 
+    def test_all_headers_are_documented(self, app):
```

Comment:
We could put this in pagination tests and make the name reflect the fact that we're checking other headers are not dropped as a non-reg test for this specific issue.

I'd be happy to enlarge the scope to check all sorts of combination and leave the test here. But this is not what we do here and I'm not sure it's worth trying.

For now, I'd move this to `test_pagination.py` and call it something like `test_pagination_doc_preservers_other_headers_doc`.

Reply:
Funilly enough, I first had that test in `test_pagination.py`, but then moved it because I felt it better fit in `test_spec.py` since it was about documentation. In any case, I moved and renamed it.